ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,274 @@
+
+        Args:
+            node: CephAdmin object
+        """
+        self.rados_obj = RadosOrchestrator(node=node)
+        self.mon_obj = MonitorWorkflows(node=node)
+        self.cluster = node.cluster
+        self.config = node.config
+        self.client = node.cluster.get_nodes(role="client")[0]
+        self.cluster_nodes = node.cluster.get_nodes()
+
+    def add_mgr_service(self, **kwargs) -> bool:
+        """
+        Method to add the mgr service from the given host
+        Args:
+            kwargs: the accepted KW args for the method are:
+                host: Cluster object of the host from where mgr needs to be Added
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mgr service exists before removing
+        host = kwargs.get("host")
+        hostname = host.hostname
+
+        if self.check_mgr_exists_on_host(host=hostname):
+            log.info(
+                f"MGR daemon already present on the host {hostname}. Returning Pass"
+            )
+            return True
+
+        log.debug(f"Adding mgr daemon on the host : {hostname}")
+
+        # Adding lable to host
+        labels = self.mon_obj.get_host_labels(host=hostname)
+        if "mgr" not in labels:
+            cmd = f"ceph orch host label add {hostname} mgr"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            time.sleep(10)
+            log.debug(f"Added mgr label for the host : {hostname}")
+
+        # Adding host as mgr
+        cmd = "ceph orch apply mgr --placement='label:mgr'"
+        self.client.exec_command(sudo=True, cmd=cmd)
+        time.sleep(10)
+
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        while datetime.datetime.now() < endtime:
+            if self.check_mgr_exists_on_host(host=hostname):
+                log.info(f"Mgr {hostname} is add to the cluster.")
+                return True
+            log.info(
+                "Mgr daemon still not added.sleeping for 5 seconds and checking again"
+            )
+            time.sleep(5)
+        return False
+
+    def remove_mgr_service(self, host) -> bool:
+        """
+        Method to remove the mgr service from the given host
+
+        Args:
+            host: name of the host from where mgr needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        mgr_daemon_list = self.get_mgr_daemon_list()
+        new_mgr_daemon = ""
+        for node in mgr_daemon_list:
+            if host.hostname in node:
+                new_mgr_daemon = node.strip()
+                break
+
+        host_name = host.hostname
+        if not self.check_mgr_exists_on_host(host=host_name):
+            log.info(f"MGR daemon not present on the host {host_name}. Returning Pass")
+            return True
+
+        # setting mgr daemon unmanaged to true
+        unmanaged_cmd_true = " ceph orch apply mgr  --unmanaged=true"
```

Comment:
Applying unmanaged flag like this changes the deployment strategy. This needs to be changed.

cluster outputs: 
``` ]# ceph orch ls mgr --export -f json

[{"placement": {"label": "mgr"}, "service_name": "mgr", "service_type": "mgr"}]

[root@ceph-pdhiran-w420vn-node1-installer ~]# ceph orch apply mgr --unmanaged=true
Scheduled mgr update...

[root@ceph-pdhiran-w420vn-node1-installer ~]# ceph orch ls mgr --export -f json
[{"placement": {"count": 2}, "service_name": "mgr", "service_type": "mgr", "unmanaged": true}]

[root@ceph-pdhiran-w420vn-node1-installer ~]# ceph orch apply mgr --unmanaged=false
Scheduled mgr update...

[root@ceph-pdhiran-w420vn-node1-installer ~]# ceph orch ls mgr --export -f json
[{"placement": {"count": 2}, "service_name": "mgr", "service_type": "mgr"}]

```

The correct way to apply unmanaged flag to the service would be to export the service spec, and then add unmanaged flag.

Reply:
I changed the command to manage and unmanaged the daemon.
Command used - ceph orch set-unmanaged mgr and ceph orch set-managed mgr

Reply:
I still see the old implementation for setting managed & unmanaged. And also, a new module is suggested for this.

Reply:
Modified the logic and added the mgr with the "ceph orch apply mgr "{no_mgrs} {new_mgr_list}"